src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js

```diff
@@ -16,54 +16,112 @@
 
 goog.module('finscholar.homepagecontroller');
 
+const GoogDom = goog.require('goog.dom');
+const GoogSoy = goog.require('goog.soy');
 const JsactionActionFlow = goog.require('jsaction.ActionFlow');
 const JsactionDispatcher = goog.require('jsaction.Dispatcher');
 const JsactionEventContract = goog.require('jsaction.EventContract');
-const GoogDom = goog.require('goog.dom');
-const GoogSoy = goog.require('goog.soy');
 const {homepage} = goog.require('finscholar.homepagecontroller.templates');
+const {CollegeListView} = goog.require('finscholar.collegelistview');
+const {CollegePageView} = goog.require('finscholar.collegepageview');
+const {ErrorPageView} = goog.require('finscholar.errorpageview');
 const {PageController} = goog.require('pagecontroller');
+const {ScholarshipListView} = goog.require('finscholar.scholarshiplistview');
+const {ScholarshipPageView} = goog.require('finscholar.scholarshippageview');
 
 /**
  * Class for the home page controller.
  * @public
  */
 class HomePageController extends PageController {
-  constructor() {
+
+  /** 
+   * @constructor
+   * @param {!Element} container The HTML div where the main frame is rendered.
+   */
+  constructor(container) {
     super();
+    /** @private @const {!Element} */
+    this.container_ = container;
+    /** @private @const {!JsactionEventContract} */
     this.eventContract_ = new JsactionEventContract();
+    /** @private @const {!JsactionDispatcher} */
+    this.dispatcher_ = new JsactionDispatcher();
+    /** @private @const {!function(): undefined} */
+    this.bindedNavbarOnclickHandler_ = this.handleNavbarOnclickEvent_.bind(this);
+    this.initJsaction_();
+    /** @private {int} */
+    this.navbarPageIndex_ = 0;
+    /** @private @const {!Array} */
```

Comment:
Specify what type this array holds

Reply:
Yeah just changed it. I can't believe the compiler doesn't throw error in this case.